sphinx_intl/catalog.py

```diff
@@ -37,7 +37,7 @@ def dump_po(filename, catalog, line_width=76):
 
     # Because babel automatically encode strings, file should be open as binary mode.
     with io.open(filename, 'wb') as f:
-        pofile.write_po(f, catalog, line_width)
+        pofile.write_po(f, catalog, line_width, sort_output=sort_output, sort_by_file=sort_by_message_location)
```

Comment:
Memo

https://babel.pocoo.org/en/latest/api/messages/pofile.html#babel.messages.pofile.write_po

* `sort_output` – whether to sort the messages in the output by msgid
* `sort_by_file` – whether to sort the messages in the output by their locations

defaults are False: `sort_output=False, sort_by_file=False`